### PR TITLE
boards: Add initial Gardena smart gateway radio module support

### PR DESCRIPTION
**Pull request by rettichschnidi**
This adds initial support for the Gardena smart gateway radio module. The module is used in all gateways of its [smart system lineup](https://www.gardena.com/int/products/smart-system/smart-system).

Once this is merged, support for SPI #78696, GPIO HD #78695 and more will follow. AES HW support (#78694 ) is already under way as it can be tested using the sim3u1xx_dk board.

This PR has been verified to pass the following testing:
```
$ twister --platform sgrm --device-testing --hardware-map ../../zephyr-hw-map-sgrm-.yml
INFO    - Using Ninja..
INFO    - Zephyr version: v3.7.0-3888-g3113fc33de93
INFO    - Using 'zephyr' toolchain.
INFO    - Building initial testsuite list...
INFO    - Writing JSON report <snip>/testplan.json

Device testing on:

| Platform   |        ID | Serial device            |
|------------|-----------|--------------------------|
| sgrm       | 269401108 | /dev/ttyUSB-sgrm-console |

INFO    - JOBS: 24
INFO    - Adding tasks to the queue...
INFO    - Added initial list of jobs to queue
INFO    - Total complete: 2878/2878  100%  skipped: 2529, failed:    0, error:    0
INFO    - 3083 test scenarios (2878 test instances) selected, 2529 configurations skipped (2409 by static filter, 120 at runtime).
INFO    - 349 of 2878 test configurations passed (100.00%), 0 failed, 0 errored, 2529 skipped with 0 warnings in 2036.05 seconds
INFO    - In total 2269 test cases were executed, 14129 skipped on 1 out of total 1 platforms (100.00%)
INFO    - 349 test configurations executed on platforms, 0 test configurations were only built.

Hardware distribution summary:

| Board   |        ID |   Counter |   Failures |
|---------|-----------|-----------|------------|
| sgrm    | 269401108 |       349 |          0 |
INFO    - Saving reports...
INFO    - Writing JSON report <snip>/twister.json
INFO    - Writing xunit report <snip>/twister.xml...
INFO    - Writing xunit report <snip>/twister_report.xml...
INFO    - Run completed
```

### CODE REVIEW

File: boards/gardena/sgrm/doc/index.rst
Comment (nordicjm) [inline boards/gardena/sgrm/doc/index.rst:35]: *board target, plus put the name as used with west build with \`\` around it

File: boards/gardena/sgrm/doc/sgrm.webp
Comment (nordicjm) [inline boards/gardena/sgrm/doc/sgrm.webp:1]: Reduce width of image, this is larger than my entire screen

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: boards/gardena/sgrm/doc/sgrm.webp
Comment (rettichschnidi) [reply boards/gardena/sgrm/doc/sgrm.webp:1]: It takes like a quarter of mine. :P

More seriously: The rendered documentation (https://builds.zephyrproject.io/zephyr/pr/79401/docs/boards/gardena/sgrm/doc/index.html) makes it fit nicely, regardless of image size. Also, scaling it further down makes loses some details (like the SiM3U top markings).

What would be a good max size? Are there some general rules, except for the < 100kB?

Comment (rettichschnidi) [reply boards/gardena/sgrm/doc/sgrm.webp:1]: Seems like most webp pictures are between 500 and 1000px wide.

For this reason, I have changed mine to 1000x749 px.

File: boards/gardena/sgrm/doc/index.rst
Comment (rettichschnidi) [reply boards/gardena/sgrm/doc/index.rst]: Fixed (I think)

**Comment by rettichschnidi** [comment]
@jerome-pouiller @jhedberg Adding you because this board is based an a SiM3U and because I'd love to get this in for Zephyr 4.0. :smile:

**Review by jhedberg — DISMISSED** [review]

**Review by jerome-pouiller — DISMISSED** [review]
A clean PR. Nothing to report.

**Review by nordicjm — DISMISSED** [review]

**Comment by rettichschnidi** [comment]
@teburd Any chance this can be merged for Zephyr 4.0? 🙏

**Comment by jhedberg** [comment]
> @teburd Any chance this can be merged for Zephyr 4.0? 🙏

@rettichschnidi I'm not sure @teburd is the most appropriate assignee, since this is basically just a Gardena board-specific change. Since you're the maintainer for such platforms I added you as a second assignee (and since you're the author this counts as an implicit approval)

File: MAINTAINERS.yml
Comment (aescolar) [inline MAINTAINERS.yml:3529]: This user does not seem to be a project contributor.
Please file a "Contributor Nomination" issue.
Otherwise it is not possible to assign @M1cha to reviews, and so forth

File: boards/gardena/sgrm/board.yml
Comment (kartben) [inline boards/gardena/sgrm/board.yml:6]:
```suggestion
  name: sgrm
  full_name: SiM3U radio module
```

But it looks as if "Smart Garden Radio Module SiM3U" might be better as it expands/explains the sgrm identifier (which tbh I'm not sure wouldn't be better with sim3u in it?)

File: boards/gardena/sgrm/doc/index.rst
Comment (kartben) [inline boards/gardena/sgrm/doc/index.rst:95]:
```suggestion
.. code-block:: shell
```
https://docs.zephyrproject.org/latest/contribute/documentation/guidelines.html#code-blocks-and-command-examples
Please check other blocks too

**Review by kartben — CHANGES_REQUESTED** [review]

File: MAINTAINERS.yml
Comment (rettichschnidi) [reply MAINTAINERS.yml:3529]: Sent him an email, waiting for a reply.

File: boards/gardena/sgrm/board.yml
Comment (rettichschnidi) [reply boards/gardena/sgrm/board.yml:6]: No other flavors exists. Went with "smart garden radio module".

File: boards/gardena/sgrm/doc/index.rst
Comment (rettichschnidi) [reply boards/gardena/sgrm/doc/index.rst]: fixed

**Comment by rettichschnidi** [comment]
Just updated the PR. Unfortunatelly, I had to rebase due to the `full_name` addition.

**Review by jhedberg — DISMISSED** [review]

File: boards/gardena/sgrm/board.yml
Comment (kartben) [reply boards/gardena/sgrm/board.yml:6]: Not convinced about the all lower case but I take it that it might not be clear to folks what the purpose of this property is and that capitalization matters and will make sure to update the board porting template.
https://builds.zephyrproject.io/zephyr/pr/79401/docs/boards/index.html#vendor=gardena

**Review by kartben — DISMISSED** [review]

File: boards/gardena/sgrm/board.yml
Comment (rettichschnidi) [reply boards/gardena/sgrm/board.yml:6]: Adapted as suggested. Not just here, but everywhere:

```console
$ rg -i "Smart Garden Radio Module"
boards/gardena/sgrm/sgrm.yaml
6:name: GARDENA Smart Garden Radio Module

boards/gardena/sgrm/sgrm.dts
12:     model = "GARDENA Smart Garden Radio Module";

boards/gardena/sgrm/board.yml
7:  full_name: Smart Garden Radio Module

boards/gardena/sgrm/doc/index.rst
3:GARDENA Smart Garden Radio Module
```

Comment (kartben) [reply boards/gardena/sgrm/board.yml:6]: Perfect - thank you, sir!

**Review by kartben — APPROVED** [review]

**Comment by rettichschnidi** [comment]
I think everything is resolved now. For this reason, I'll re-request the reviews for (hopefully) one last time. Thanks a lot!

**Review by M1cha — APPROVED** [review]

**Pull request merged**
